src/crypto/algorithms/megolm.ts

```diff
@@ -277,37 +279,37 @@ class MegolmEncryption extends EncryptionAlgorithm {
         // Updates `session` to hold the final OutboundSessionInfo.
         //
         // returns a promise which resolves once the keyshare is successful.
-        const prepareSession = async (oldSession: OutboundSessionInfo) => {
-            session = oldSession;
```

Comment:
Not sure how much I agree, because now `returnSession` returns `OutboundSessionInfo | undefined` which muddies the waters since `setupPromise` says `OutboundSessionInfo | null` - we know we'll never call `returnSession` before `prepareSession` right now but that is not ensured.

Reply:
Hmm, i'll give this another closer look then, see what i can do to assure better correctness